src/climatevision/generator/makeentries.py

```diff
@@ -12,6 +12,10 @@ def make_entries(
     assert year_baseline in YEAR_BASELINE_CHOICES
     assert year_target in YEAR_TARGET_CHOICES
 
+    if ags not in data.ags_master():
+        print(f"ags '{ags}' was not found in the master CSV, please check your entry.")
+        exit(1)
+
```

Comment:
Vielleicht kannst du nochmal versuchen eine Lösung zu finden, die sich am Bestandscode orientiert.
z.B. in etwa so: assert ags in data.ags_master()

Das führt wahrscheinlich noch nicht im Devtool zu einer einfachen Fehlermeldung. Ich habe gerade nochmal geschaut, Die anderen Fehlermeldungen kommen wahrscheinlich aus arguments.py. Vielleicht kannst du sonst da nochmal rein schauen.

Reply:
Ist bisschen komplizierter mit den `ags`, oder bekomme ich alle "korrekten" Inputs noch anders?
Ich hätte es jetzt erstmal so gelöst:
```python
import argparse
from typing import Any

from climatevision.generator import RefData
from climatevision.generator.years import (
    YEAR_BASELINE_CHOICES,
    YEAR_BASELINE_DEFAULT,
    YEAR_BASELINE_HELP,
    YEAR_REF_CHOICES,
    YEAR_REF_DEFAULT,
    YEAR_REF_HELP,
    YEAR_TARGET_CHOICES,
    YEAR_TARGET_DEFAULT,
    YEAR_TARGET_HELP,
)


class DynamicChoicesAction(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        year_ref = getattr(namespace, "year_ref", 2018)  # Default to 2018 if not set
        ags_master = RefData.load(int(year_ref)).ags_master()
        if values not in ags_master:
            raise argparse.ArgumentError(
                self,
                f"Invalid choice: {values!r}. Valid choices include: {', '.join(list(ags_master)[:10])}, ...",
            )

        setattr(namespace, self.dest, values)


def add_ags_argument(parser: Any):
    parser.add_argument("-ags", default="03159016", action=DynamicChoicesAction)
```